src/Ema/Route/Generic/Example.hs

```diff
@@ -100,3 +104,47 @@ instance EmaSite R2 where
 
 mainConst :: IO ()
 mainConst = Ema.runSite_ @R2 ()
+
+-- ---
+-- Ensure deriveIsRoute works with (partial) subroutes
+-- --
+
+data M3
+  = M3
+  deriving stock (Eq, Show, Generic)
+
+data R3
+  = R3_Sub R3_SubR
+  | R3_Index
+
+data R3_SubR
+  = R3_SubR
+  deriving stock (Show, Eq, Ord, Generic)
+  deriving anyclass (SOP.Generic, SOP.HasDatatypeInfo)
+  deriving
+    (HasSubRoutes, HasSubModels, IsRoute)
+    via ( GenericRoute
+            R3_SubR
+            '[ WithSubRoutes
+                '[ FileRoute "example.html"
+                 ]
+             ]
+        )
+
+SOP.deriveGeneric ''R3
+deriveIsRoute ''R3 ''M3 $ Just [''R3_SubR, ''()]
```

Comment:
Do you remember off hand what the TH syntax is here if `M3` was, say, `Maybe M3` or `Map Text M3` instead?

Reply:
Specifically, I wonder how to use your TH in Ex03_Store.hs which doesn't use simple model types.

Also, why can't we use TH for `R3_SubR`?

Reply:
I meant to comment on this last night actually. Initially I decided to go for a design wherein the user can pass in a type list via [t||] but got to thinking that the examples were just for the simple case of wrapping simple (kind *) routes, so I didn't want to overcomplicate things at first. But if we want to support it to this extent I can definitely go back to the initial implementation.

Reply:
Yea, since GHC 9.2 is not the default in nixpkgs yet, I'd like to support the following scenario ideally:

```haskell
deriveGeneric ''ProductRoute
deriveIsRoute ''ProductRoute (Map Slug Product) $
  Just $
    ( [ FileRoute "index.html"
      , StringRoute Product Slug
      ]
    )
```

Even with GHC 9.2, it can be useful to be able to override subroute encoding this way. Ex03_Store does it for this reason in `ProductRoute` (to remove the extra `/product`/ prefix injected by generic subroutes).

We probably also want to provide two variants `deriveIsRoute ''R ''M` and `deriveIsRouteWith ''R [ WithModel M, WithSubRoutes [ FileRoute "...", ...] ]` and remove the `Maybe` from the equation.

Reply:
> We probably also want to provide two variants

Or maybe we should provide only one variant. The simplest case (unit model) would then look:

```haskell
deriveIsRoute 'R []
```

The case with a model would look a bit complex but still not bad:

```haskell
deriveIsRoute 'R [ WithModel M ]
```

Reply:
It is probably also more straightforward to implement in TH because you can take the whole type expression and slap it onto `GenericRoute`?

Reply:
```hs
SOP.deriveGeneric ''R3
deriveIsRoute ''R3 [t|
    [ WithModel M3
    , WithSubRoutes
        [ R3_SubR
        , ()
        ]
    ]
  |]
```

Does this look alright? I don't know of any way to pass a `Type` directly as an argument outside of quasiquoting it like this or some Proxy black magic.

Reply:
That doesn't look too bad to me. What does the trivial case look like -- `deriveIsRoute ''R [t| [] ]`?

Reply:
Yea

Reply:
Okay, let's go with that for now.